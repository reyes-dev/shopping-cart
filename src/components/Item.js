import React, { useState } from "react";

const Item = ({ filePath, name, incrementCartQuantity, addToCart }) => {
  // The amount of an item that the shopper wants to purchase
  const [itemCount, setItemCount] = useState(0);
  // Set amount of item to add to cart
  const incrementCount = () => {
    setItemCount((count) => count + 1);
  };
  const decrementCount = () => {
    setItemCount((count) => {
      if (count > 0) {
        return count - 1;
      } else {
        return 0;
      }
    });
  };

  return (
    <div>
      <img src={filePath} alt={name}></img>
      <li>{name}</li>
      <button onClick={incrementCount}>+</button>
      <p>{itemCount}</p>
      <button onClick={decrementCount}>-</button>
      <button
        onClick={() => {
          incrementCartQuantity();
          addToCart({ name: name, filePath: filePath, quantity: itemCount });
        }}
      >
        Add to Cart
      </button>
    </div>
  );
};

export default Item;
