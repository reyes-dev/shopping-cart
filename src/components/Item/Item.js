import React, { useState } from "react";

const Item = ({ filePath, name, incrementCartQuantity, addToCart }) => {
  // The amount of an item that the shopper wants to purchase
  const [itemCount, setItemCount] = useState(0);
  // Set amount of item to add to cart
  const incrementCount = () => {
    setItemCount((count) => count + 1);
  };
  const decrementCount = () => {
    setItemCount((count) => {
      if (count > 0) {
        return count - 1;
      } else {
        return 0;
      }
    });
  };

  return (
    <div className="item" data-testid="item-component">
      <img src={filePath} alt={name}></img>
      <li>{name}</li>
      <div className="item-counter">
        <button onClick={decrementCount}>-</button>
        <p>{itemCount}</p>
        <button onClick={incrementCount}>+</button>
      </div>
      <button
        onClick={() => {
          if (itemCount > 0) {
            incrementCartQuantity();
            addToCart({
              name: name,
              filePath: filePath,
              quantity: itemCount,
            });
            setItemCount(0);
          }
        }}
      >
        Add to Cart
      </button>
    </div>
  );
};

export default Item;
