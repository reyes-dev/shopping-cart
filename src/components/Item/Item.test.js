import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Item from "./Item";

describe("Item component", () => {
  it("renders 0", () => {
    const { container } = render(<Item />);
    expect(container).toMatchSnapshot();
  });

  it("renders 1 after button click", async () => {
    const user = userEvent.setup();

    render(<Item />);
    const button = screen.getByRole("button", { name: "+" });

    await user.click(button);

    expect(screen.getByTitle("item-count").textContent).toMatch("1");
  });

  it("renders 1 after incrementing twice, decrementing once", async () => {
    const user = userEvent.setup();

    render(<Item />);

    const incrementBtn = screen.getByRole("button", { name: "+" });
    const decrementBtn = screen.getByRole("button", { name: "-" });

    await user.click(incrementBtn);
    await user.click(incrementBtn);

    expect((await screen.findByTitle("item-count")).textContent).toMatch("2");

    await user.click(decrementBtn);

    expect((await screen.findByTitle("item-count")).textContent).toMatch("1");
  });
});
