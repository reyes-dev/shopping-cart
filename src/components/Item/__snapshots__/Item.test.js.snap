// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component renders 0 1`] = `
<div>
  <div
    class="item"
    data-testid="item-component"
  >
    <img />
    <li />
    <div
      class="item-counter"
    >
      <button>
        -
      </button>
      <p
        title="item-count"
      >
        0
      </p>
      <button>
        +
      </button>
    </div>
    <button>
      Add to Cart
    </button>
  </div>
</div>
`;
